### ISSUE TITLE
MOSQUITTO Client <unknown> has exceeded timeout, disconnecting

### ISSUE THREAD

**Issue by nedimhx** [knolleary/pubsubclient#445]
Same source code on 3 NodeMCUs as RFID reader, but one of them has always problem after some time cliend id is marked as unknown.

Why ?
3 NodeMCUs as RFID
2 NodeMCUs as senosors reader 

And every time only this one has a problem?

**Comment by knolleary** [comment]
You say they have the same source code - but have you provided a unique client id (the string you pass to the `connect` function) to each of them?

**Comment by nedimhx** [comment]
of course, in the code there is clientid generator, so clientid is uniqe.

It may be problem with internet connection.
